src/main/java/htsjdk/variant/variantcontext/writer/VariantContextWriterBuilder.java

```diff
@@ -468,68 +486,75 @@ else if (STREAM_TYPES.contains(this.outType))
      * attempt to resolve any symlinks and try again.  If that fails, and the output file exists
      * but is neither a file or directory then VCF_STREAM is returned.
      */
-    protected static OutputType determineOutputTypeFromFile(final File f) {
-        if (isBCF(f)) {
+    protected static OutputType determineOutputTypeFromFile(final Path path) {
+        if (isBCF(path)) {
             return OutputType.BCF;
-        } else if (isCompressedVCF(f)) {
+        } else if (isCompressedVCF(path)) {
             return OutputType.BLOCK_COMPRESSED_VCF;
-        } else if (isVCF(f)) {
+        } else if (isVCF(path)) {
             return OutputType.VCF;
         }
         else {
             // See if we have a special file (device, named pipe, etc.)
-            final File canonical = new File(IOUtil.getFullCanonicalPath(f));
-            if (!canonical.equals(f)) {
-                return determineOutputTypeFromFile(canonical);
+            try {
+                // Try to resolve symlinks etc.
```

Comment:
It's not entirely clear to me, but I think `path.toRealPath()` is essentially equivalent to `getFullCanonicalPath`

Reply:
OK, I switched this code to one with Path and using toRealPath, hopefully that's what you had in mind!

Reply:
`.toRealPath` throws `java.nio.file.NoSuchFileException` when passed `/dev/stdout`, I added a special case for that.